src/autotype/mac/AutoTypeMac.cpp

```diff
@@ -470,6 +479,21 @@ OSStatus AutoTypePlatformMac::hotkeyHandler(EventHandlerCallRef nextHandler, Eve
 {
     Q_UNUSED(nextHandler);
 
+    // Determine if the user has given proper permissions to KeePassXC to perform Auto-Type
+    if (!accessibilityChecked) {
+        if (macUtils()->enableAccessibility() && macUtils()->enableScreenRecording()) {
```

Comment:
So despite reverting to Carbon, we still need these permissions, right?

Reply:
Yah carbon had nothing to do with the actual typing (accessibility) and window title matching (screen recording)